lib/rack/cors.rb

```diff
@@ -14,17 +14,17 @@ def initialize(app, opts={}, &block)
       @app = app
       @debug_mode = !!opts[:debug]
 
-      if logger = opts[:logger]
-        if logger.respond_to? :call
+      if opts[:logger]
+        if opts[:logger].respond_to? :call
           @logger_proc = opts[:logger]
         else
-          @logger = logger
+          @logger = opts[:logger]
         end
       end
 
       if block_given?
         if block.arity == 1
-          block.call(self)
+          yield self
         else
           instance_eval(&block)
```

Comment:
Glad to be able to share that little trick! :smile:

Let me see if `Proc.new()` will have the correct arity information. If it does then we should be able to get rid of `&block` in the definition.